src/hotspot/cpu/aarch64/macroAssembler_aarch64.cpp

```diff
@@ -399,6 +399,14 @@ static inline bool target_needs_far_branch(address addr) {
   return !CodeCache::is_non_nmethod(addr);
 }
 
+static inline bool target_needs_trampoline(Address entry) {
+  // Java method call target can have different compiled versions of a method: C2 or C1.
+  // If CodeCache size > 128M, such calls must have reserved trampolines for the case:
+  // C2 compiled method calls C1 compiled method.
+  return  (entry.rspec().type() == relocInfo::runtime_call_type) ? target_needs_far_branch(entry.target())
```

Comment:
Why is target_needs_far_branch() applied for runtime_call_type only?

Reply:
Updated comment to make it clear.